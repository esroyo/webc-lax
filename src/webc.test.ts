import { parse5, WebC } from './index.ts';
import { assertSnapshot } from '@std/testing/snapshot';

Deno.test('should replace slots contents and keep the positions in the document (head or body)', async (t) => {
    const page = new WebC();
    page.setContent(`<html>
  <head>
    <title>My website</title>
    <slot name="my-styles"><style>* { color: initial; }</style></slot>
  </head>
  <body>
    <slot name="my-js"></slot>
  </body>
</html>
`);
    const slots = {
        'my-styles': parse5.parseFragment('<style>* { color: red; }</style>'),
        'my-js': parse5.parseFragment(
            '<script>console.log("Hello world!");</script>',
        ),
    };
    const { html } = await page.compile({ slots });
    await assertSnapshot(t, html);
});

Deno.test('should replace slot contents if provided or use the default contents', async (t) => {
    const page = new WebC();
    page.setContent(`<html>
  <head>
    <title>My website</title>
  </head>
  <body>
    <slot name="my-js"></slot>
    <slot name="my-styles"><style>p { color: blue }</style></slot>
  </body>
</html>
`);
    const slots = {
        'my-js': parse5.parseFragment(
            '<script>console.log("Hello world!");</script>',
        ),
    };
    const { html } = await page.compile({ slots });
    await assertSnapshot(t, html);
});
