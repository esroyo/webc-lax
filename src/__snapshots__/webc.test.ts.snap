export const snapshot = {};

snapshot[`should replace slots contents and keep the positions in the document (head or body) 1`] = `
'<!doctype html>
<html>
<head>
    <title>My website</title>
    <style>* { color: red; }</style>
  </head>
  <body>
    <script>console.log("Hello world!");</script>
  

</body>
</html>'
`;

snapshot[`should replace slot contents if provided or use the default contents 1`] = `
'<!doctype html>
<html>
<head>
    <title>My website</title>
  </head>
  <body>
    <script>console.log("Hello world!");</script>
    <style>p { color: blue }</style>
  

</body>
</html>'
`;
